From 6d3cb8464777e142e2d30f14286a0e09babf9ab1 Mon Sep 17 00:00:00 2001
From: Li Jun <jun.li@nxp.com>
Date: Mon, 5 Jun 2017 18:29:43 +0800
Subject: [PATCH 2232/5242] MLK-16013-7 usb: dwc3: gadget: increase timeout
 count for ep cmd

commit  a1d1fa2220288266fd0484055517a82c12dd407b from
https://source.codeaurora.org/external/imx/linux-imx.git

Increase the timeout value for wait ep command complete, this is temp
solution to workaround it but no harm any good cases.

Reviewed-by: Peter Chen <peter.chen@nxp.com>
Signed-off-by: Li Jun <jun.li@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/usb/dwc3/gadget.c |    6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/drivers/usb/dwc3/gadget.c b/drivers/usb/dwc3/gadget.c
index 5f6ccbd..82e117c 100644
--- a/drivers/usb/dwc3/gadget.c
+++ b/drivers/usb/dwc3/gadget.c
@@ -269,7 +269,11 @@ int dwc3_send_gadget_ep_cmd(struct dwc3_ep *dep, unsigned cmd,
 {
 	const struct usb_endpoint_descriptor *desc = dep->endpoint.desc;
 	struct dwc3		*dwc = dep->dwc;
-	u32			timeout = 1000;
+				/*
+				 * FIXME need check why 500 times check
+				 * is not enough.
+				 */
+	u32			timeout = 20000;
 	u32			reg;
 
 	int			cmd_status = 0;
-- 
1.7.9.5

